--- src/lib/Libutils/u_threadpool.c.orig	2013-12-04 16:15:07.000000000 -0700
+++ src/lib/Libutils/u_threadpool.c	2013-12-04 16:17:50.000000000 -0700
@@ -80,6 +80,7 @@
 
 #include <errno.h>
 #include <signal.h>
+#include <sys/time.h>
 #include <time.h>
 #include "threadpool.h"
 #include "utils.h"
@@ -206,6 +207,7 @@
   void         *arg;
   tp_work_t    *mywork;
   tp_working_t  working;
+  struct timeval tv;
 
   struct timespec   ts;
 
@@ -241,7 +243,9 @@
         }
       else
         {
-        clock_gettime(CLOCK_REALTIME,&ts);
+         gettimeofday(&tv, NULL);
+         ts.tv_sec = tv.tv_sec;
+         ts.tv_nsec = 0;
         ts.tv_sec += request_pool->tp_max_idle_secs;
         rc = pthread_cond_timedwait(&request_pool->tp_waiting_work,&request_pool->tp_mutex,&ts);
 
